student/src/main/java/tds/student/tdslogger/ApplicationShutdownLogger.java

```diff
@@ -0,0 +1,31 @@
+package tds.student.tdslogger;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextClosedEvent;
+import org.springframework.stereotype.Component;
+
+import static tds.student.tdslogger.StudentEventLogger.StudentLogEvent.APP_SHUTDOWN;
+
+@Component
+public class ApplicationShutdownLogger implements ApplicationListener<ContextClosedEvent> {
+
+  private StudentEventLogger logger;
+  boolean logged = false;
+
+  @Autowired
+  public ApplicationShutdownLogger(@Qualifier("integrationObjectMapper") final ObjectMapper objectMapper) {
+    logger = new StudentEventLogger(objectMapper);
+  }
+
+  @Override
+  public void onApplicationEvent(final ContextClosedEvent event) {
+    // This event fires multiple times - we only want to log the first occurrence.
+    if (!logged) {
```

Comment:
This class is a singleton (by default all spring beans are) so this will only be logged once for an entire applications lifecycle.  To get it to log again it'll require a restart of the application server

Not sure if that is the intention.  If that is the intention I would appreciate a little more robust comments in here explaining what this is doing and why it only logs once.  For future maintainers that do not know anything about the logging work.